lightning/src/ln/msgs.rs

```diff
@@ -88,6 +88,18 @@ pub struct ErrorMessage {
 	pub data: String,
 }
 
+/// An warning message to be sent or received from a peer
+#[derive(Clone, Debug, PartialEq)]
+pub struct WarningMessage {
+	/// The channel ID involved in the warning
+	pub channel_id: [u8; 32],
+	/// A possibly human-readable warning description.
+	/// The string should be sanitized before it is used (e.g. emitted to logs
+	/// or printed to stdout).  Otherwise, a well crafted error message may trigger a security
+	/// vulnerability in the terminal emulator or the logging subsystem.
```

Comment:
Should we recommend further to rate-limit `WarningMessage` logging to avoid disk I/O DoS ? Core does have `-maxreceivebuffer/maxsendbuffer`. Though I think it's part of a wider reasoning about our DoS mitigations strategy.

Reply:
Yes, there's an issue for this somewhere. I'd love to make the warning messages `log_warn` or so, but for DoS reasons its only `log_debug`. I *think* a reasonable metric or something would be "more than `log_debug` must not be DoS vulnerable", but I'm sure there are issues right now.